modules/react/layout/stories/Grid.stories.mdx

```diff
@@ -0,0 +1,94 @@
+### UI Layout Example
+
+This example showcases what a basic UI could look like utilzing Grid. In the below example, we have
+created four elements (`Header`, `SideBar`, `BodyContent` and `Footer`). By giving each element a
+`gridArea` name, the name can be used to organize the UI however you want. If you want more info on
+how `gridArea` works with `gridTemplateAreas`,
+[see here](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-areas).
+
+<ExampleCodeBlock code={UIExample} />
+
+### Rows and Columns
+
+You can nest `Grid` components to build layouts with rows and columns. You can also nest `Grid`
+components within `<Grid.Item/>` components for even more possibilities.
+
+In the below example, we have a `Grid` container. It contains nine elements with one of them being
+grey. This is a `<Grid.Item/>` component and has access to child specific props, the `<Grid/>`
+component only has access to container props. This is to show what the `<Grid.Item/>` component
+can&nbsp;do.
+
+In the layout, the `<Grid.Item/>` component (grey element), has a `Row` and `Col` button. These
+buttons are interactive and if you click the `Row` button, it will move the `<Grid.Item/>` element
+between the rows and if you click the `Col` button, it will move it through the columns. These
+buttons are updating `gridRowStart`(`grid-row-start` css grid property) and
+`gridColumnStart`(`grid-column-start` css grid property). This is to showcase the child specific
+props for the `<Grid.Item/>` component.
+
+<ExampleCodeBlock code={GridLayoutInteractive} />
+
+**⚠️This is NOT best practice for accessibility reasons and we suggest against following this. This
+is merely to showcase the child specific props for the `<Grid.Item/>` component.⚠️**
+
+In the second example below, we have a `Grid` container. It contains two rows with one having seven
+elements and the other has two elements. This example is to show how a user can use the
+`<Grid.Item/>` component to wrap a `<Grid/>` component. This would allow a user to have access to
+the child props of `<Grid.Item/>` to place a layout where needed. The row with elements 3 - 7 is put
+before the 1 & 2 elements since that layout is wrapped by the `<Grid.Item/>` component.
+
+<ExampleCodeBlock code={GridLayout} />
+
+### Grid vs. Box
+
+`Grid` is built with `Box` and has access to all `BoxProps`. The main differences between `Box` and
+`Grid` are:
+
+- `Grid` is set to `display: grid` & `inline-grid` by default.
+- The `display` prop is limited to `grid` & `inline-grid` .
+- `Grid` has access to grid container properties.
+- `Grid.Item` has access to grid child properties.
+
+In short, use `Grid` when you need a grid container to build layouts with CSS Grid, and use `Box`
+when you need a simple container. It's also important to note that `Grid` has no real opinion about
+how to build layouts outside of the CSS Grid spec. Its unopinonated nature allows it to be composed
+into many different components.
+
+## Grid Props
```

Comment:
Let's follow the [Compound Component MDX Structure](https://github.com/Workday/canvas-kit/discussions/1131) that James described.

```suggestion
## Components

### Grid

Grid is a container component for creating two-dimensional layouts with CSS Grid. It has special style props that map to CSS Grid style properties to provide a common, ergonomic API for building layouts.
```

Following that, we can add small `Usage` and `Props` sections. GitHub doesn't like MD code blocks in suggestions, so I'll add them below.


### Usage

```tsx
<Grid grid gridTemplateColumns="1fr 2fr 1fr" gridGap={space.s}>
  <div>Implicit grid item 1</div>
  <div>Implicit grid item 2</div>
  <div>Implicit grid item 3</div>
</Grid>
```

### Props